source/development/coredump.rst

```diff
@@ -0,0 +1,195 @@
+
+#. **Enable Core Dump Collection**:
+    Enable core dump collection by setting the ``Storage=`` option in the systemd ``coredump.conf`` file. You can set it to ``external`` to store core dumps externally, or ``none`` to disable core dump collection altogether.
+
+    .. code-block:: console
+
+        systemctl edit systemd-coredump
+
+    Add the following lines to the editor that opens:
+
+    .. code-block:: console
+
+        [Coredump]
+        Storage=external
+
+#. **Recommended** – Set a size limit for core dump files. For example, 2 GB.
+
+    .. code-block:: console
+
+        ProcessSizeMax=2G
+
+#. **Restart systemd-coredump Service**:
+    After making changes to the configuration, restart the systemd-coredump service for the changes to take effect:
+
+    .. code-block:: console
+
+        systemctl restart systemd-coredump
+
+By following these steps, you can set up core dump generation using systemd. This allows for centralized management and configuration of core dumps across your system.
+
+To disable core dump generation, you can modify the configuration of the systemd-coredump service. Here's how you can do it:
+
+#. **Edit systemd-coredump Configuration**:
+    Open the systemd-coredump configuration file for editing:
+
+    .. code-block:: console
+
+        systemctl edit systemd-coredump
+
+#. **Add Configuration to Disable Core Dumps**:
+    Add the following lines to the editor that opens:
+
+    .. code-block:: console
+
+        [Coredump]
+        Storage=none
+
+    This configuration sets the ``Storage`` option to ``none``, effectively disabling core dump storage.
+
+#. **Restart systemd-coredump Service**:
+    After making changes to the configuration, restart the systemd-coredump service for the changes to take effect:
+
+    .. code-block:: console
+
+        systemctl restart systemd-coredump
+
+#. **Verify Configuration**:
+    Check the status of the systemd-coredump service to ensure it is running without errors:
+
+    .. code-block:: console
+
+        systemctl status systemd-coredump
+
+Once you've completed these steps, core dump generation will be disabled on your system. Any attempts to generate core dumps will not result in files being stored.
+
+Manual configuration
+^^^^^^^^^^^^^^^^^^^^
+
+Setting up core dump generation without using systemd involves configuring the operating system's core dump settings manually. Here's how you can do it on Linux systems:
+
+#. **Identify the Target Directory for Core Dumps**:
+    Choose a directory where core dump files will be generated. This directory should have sufficient disk space and appropriate permissions for the process generating core dumps to write to it.
+
+#. **Set the Core Dump Size Limit**:
+    If the current core dump size limit is insufficient, increase it using the ``ulimit`` command. For example, to set the core dump size limit to unlimited:
```

Comment:
How does the user know whether the size limit is enough? Or do we mean we recommend to set it to unlimited always?

Reply:
There is no precise answer for this, since the size of the core will depend on several factors, such as the size of the process, the point where the error occurs, etc.
The use of unlimited is common.

Reply:
We may turn this step to "use `unlimited`" as our recommendation then, regardless whether the current size limit of the user is enough.

Reply:
I only added this recommendation because I have noticed it in some publications consulted, but as has already been said, the use of unlimited is common, could we remove this warning if you prefer?

Reply:
Rather, I would remove "if the limit is insufficient" and would recommend to set the limit to unlimited, then. For example: "Set the core dump size limit to unlimited".